src/main/java/io/fabric8/maven/docker/model/ContainerDetails.java

```diff
@@ -121,6 +126,21 @@ public boolean isRunning() {
         return state.getBoolean(RUNNING);
     }
 
+    @Override
+    public boolean isHealthy() {
+        final JSONObject state = json.getJSONObject(STATE);
+        // always indicate healthy for docker hosts that do not support health checks.
+        return !state.has(HEALTH) || HEALTH_STATUS_HEALTHY.equals(state.getJSONObject(HEALTH).getString(STATUS));
+    }
+
+    @Override
+    public String healthcheckTests() {
+        if (!json.getJSONObject(CONFIG).has(HEALTHCHECK)) {
+            return null;
+        }
+        return json.getJSONObject(CONFIG).getJSONObject(HEALTHCHECK).getJSONArray(TEST).join(", ");
```

Comment:
I think its worth the extra round trip (maybe it could be even cached somewhere else ?), because although the plugin won't break if there are no health-check support I think its important to inform the user that the health-check doesn't work because he is running against an too-old Docker daemon. 

So the user configuration can fail for two case: Either no health-check has been provided **or** the Docker daemon doesn't support health-check. We should report that differently. In the first case its a configuration error (doesn't make sense to wait in this case) in the later it's because a potentially supported configuration is run in an non-compatible environment.